posts/2022-02-08-feature-generation-tech-preview.adoc

```diff
@@ -0,0 +1,130 @@
+
+== Dev mode with automatic generation of features
+
+For Maven projects, specify the Liberty Maven Plugin with version `3.5.2-SNAPSHOT` and its Sonatype repository in your `pom.xml` file:
+[source,xml]
+----
+    <pluginRepositories>
+        <pluginRepository>
+            <id>sonatype</id>
+            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+        </pluginRepository>
+    </pluginRepositories>
+    <repositories>
+        <repository>
+            <id>sonatype</id>
+            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
+        </repository>
+    </repositories>
+
+<plugin>
+    <groupId>io.openliberty.tools</groupId>
+    <artifactId>liberty-maven-plugin</artifactId>
+    <version>3.5.2-SNAPSHOT</version>
+</plugin>
+----
+
+For Gradle projects, specify the Liberty Gradle Plugin with version `3.3.1-SNAPSHOT` and its Sonatype repository in your `build.gradle` file:
+[source, groovy]
+----
+apply plugin: 'liberty'
+
+buildscript {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        maven {
+            url "https://plugins.gradle.org/m2"
+            url "https://oss.sonatype.org/content/repositories/snapshots"
+        }
+    }
+    dependencies {
+        classpath 'io.openliberty.tools:liberty-gradle-plugin:3.3.1-SNAPSHOT'
+    }
+}
+----
+
+Start dev mode as usual. Automatic generation of features is on by default in the technology preview (snapshot release).
+Maven: `mvn liberty:dev`
+Gradle: `gradle libertyDev`
+
+image::/img/blog/liberty-dev-startup.png[Liberty dev mode startup with automatic generation of features on,width=70%,align="center"]
+
+As you make code changes to your application, dev mode will monitor your API usage in your class files. Then, it will automatically generate and install a new Liberty feature if needed. The feature list will be generated in a new file `configDropins/overrides/generated-features.xml` in the configuration directory of your project.
+
+image::/img/blog/liberty-generate-features-increment.png[Liberty dev mode generating features after code change,width=70%,align="center"]
+[source, xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<server>
+    <!--This file was generated by the Liberty Maven Plugin and will be overwritten on subsequent runs of the liberty:generate-features goal.
+ It is recommended that you do not edit this file and that you commit this file to your version control.-->
+    <featureManager>
+        <!--The following features were generated based on API usage detected in your application-->
+        <feature>cdi-2.0</feature>
+        <feature>mpHealth-3.0</feature>
+        <feature>mpMetrics-3.0</feature>
+        <feature>jaxrs-2.1</feature>
+    </featureManager>
+</server>
+----
+
```

Comment:
Can we label this section as "Example `generated-features.xml`" just to make it a little more clear?